ghost/milestone-emails/lib/MilestonesEmailService.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @typedef {import('./MilestonesAPI')} MilestonesAPI
+ * @typedef {import('./Milestone')} Milestone
+ */
+
+/**
+ * @template Model
+ * @typedef {import('./MilestonesAPI')} <Model>
+ */
+
+/**
+ * @typedef {Object} IQueries
+ * @prop {() => Promise<number>} getMembersCount
+ * @prop {() => Promise<Object>} getARR
+ * @prop {() => Promise<boolean>} hasImportedMembersInPeriod
+ */
+
+module.exports = class MilestonesEmailService {
+    /** @type {import('./MilestonesAPI')} */
+    #api;
+
+    /** @type {Function} */
+    #mailer;
+
+    /** @type {Object} */
+    #config;
+
+    /** @type {IQueries} */
+    #queries;
+
+    /** @type {string} */
+    #defaultCurrency;
+
+    /**
+     * @param {object} deps
+     * @param {Function} deps.mailer
+     * @param {MilestonesAPI} deps.api
+     * @param {Object} deps.config
+     * @param {IQueries} deps.queries
+     * @param {string} deps.defaultCurrency
+     */
+    constructor(deps) {
+        this.#mailer = deps.mailer;
+        this.#api = deps.api;
+        this.#config = deps.config;
+        this.#queries = deps.queries;
+        this.#defaultCurrency = deps.defaultCurrency;
+    }
+
+    /**
+     *
+     * @param {Array} goalValues
+     * @param {number} current
+     *
+     * @returns {Array}
+     */
+    _getMatchedMilestone = (goalValues, current) => {
+        // return highest suitable milestone
+        return goalValues.filter(value => current >= value)
+            .sort((a, b) => b - a)[0];
+    };
+
+    /**
+     *
+     * @param {number} milestone
+     * @param {'arr'|'members'} type
+     * @param {boolean} hasMembersImported
+     *
+     * @returns {Promise<Milestone>}
+     */
+    _saveMileStoneAndSendEmail = async (milestone, type, hasMembersImported) => {
+        const milestoneData = {
+            type,
+            value: milestone
+        };
+
+        if (type === 'arr') {
+            milestoneData.currency = this.#defaultCurrency;
+        }
+
+        // Two cases in which we don't want to send an email
+        // 1. There has been an import of members within the last week
+        // 2. The last email has been sent less than two weeks ago
+        const shouldSendEmail = await this.#api.shouldSendEmail();
+
+        if (shouldSendEmail && !hasMembersImported) {
+            // TODO: hook up GhostMailer or use StaffService and trigger event to send email
+            // await this.#mailer.send({
```

Comment:
This is a step to decide next: use GhostMailer directly or utilise the StaffService and send an event 🤔